geomesa-accumulo/geomesa-accumulo-datastore/src/test/scala/org/locationtech/geomesa/process/tube/TubeSelectProcessTest.scala

```diff
@@ -259,6 +259,50 @@ class TubeSelectProcessTest extends Specification {
 
       results.size mustEqual 20
     }
+
+    "should properly handle geometries crossing the IDL" in {
+      val sftName = "tubeTestType"
+      val sft = SimpleFeatureTypes.createType(sftName, s"type:String,$geotimeAttributes")
+
+      val ds = createStore
+
+      ds.createSchema(sft)
+      val fs = ds.getFeatureSource(sftName)
+
+      val featureCollection = new DefaultFeatureCollection(sftName, sft)
+
+      List("c", "d").foreach { name =>
+        List(1, 2, 3, 4).zip(List(-178, -179, 179, 178)).foreach { case (i, lat) =>
+          val sf = AvroSimpleFeatureFactory.buildAvroFeature(sft, List(), name + i.toString)
+          sf.setDefaultGeometry(WKTUtils.read(f"POINT($lat%d 0)"))
+          sf.setAttribute(DefaultDtgField, new DateTime(f"2011-01-01T00:0$i%d:00Z", DateTimeZone.UTC).toDate)
+          sf.setAttribute("type", name)
+          sf.getUserData()(Hints.USE_PROVIDED_FID) = java.lang.Boolean.TRUE
+          featureCollection.add(sf)
+        }
+      }
+
+      // write the feature to the store
+      val res = fs.addFeatures(featureCollection)
+
+      // tube features
+      val tubeFeatures = fs.getFeatures(CQL.toFilter("type = 'c'"))
+
+      // result set to tube on
+      val features = fs.getFeatures(CQL.toFilter("type <> 'd'"))
+
+      // get back type b from tube
+      val ts = new TubeSelectProcess()
+      val results = ts.execute(tubeFeatures, features, null, 1L, 1L, 0.0, 5, "line")
```

Comment:
what happened previously? i.e. is just successfully returning enough of a check, or should we validate the results a little more?

Reply:
Addressed this by adding points that would be grabbed if the IDL were handled as it has been in the past. They are now not picked up in the resulting query, proving that the change has had an effect. Previously, if a segment crossed the IDL, the resulting geometry would wrap the wrong way around the globe resulting in a buffered line that wrapped the globe. This would grab any points that occurred in the time between points within that belt. Now, because the geometry is split at the IDL, the geometry properly handles the IDL and does not pick up points outside of the intended area.